### ISSUE TITLE
Strange warning stating that export does not exist when it does

### ISSUE THREAD

**Issue by domehead100** [webpack/webpack#7378]
# Bug report

**What is the current behavior?**

Not sure if this is a problem with a loader or webpack itself, but on a project that is using typescript, I'm getting a warning:
```javascript
    WARNING in ./React/src/components/_shared/form/form-model.ts
    17:0-39 "export 'IValidation' was not found in '_types/IValidation'"
        at HarmonyExportImportedSpecifierDependency._getErrors (C:\project\node_modules\webpack\lib\dependencies\HarmonyExportImportedSpecifierDependency.js:352:11)
        at HarmonyExportImportedSpecifierDependency.getWarnings (C:\project\node_modules\webpack\lib\dependencies\HarmonyExportImportedSpecifierDependency.js:299:15)
        at Compilation.reportDependencyErrorsAndWarnings (C:\project\node_modules\webpack\lib\Compilation.js:985:24)
        at Compilation.finish (C:\project\node_modules\webpack\lib\Compilation.js:817:9)
        at hooks.make.callAsync.err (C:\project\node_modules\webpack\lib\Compiler.js:479:17)
        at _done (eval at create (C:\project\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:9:1)
        at _err0 (eval at create (C:\project\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:20:22)
        at _addModuleChain (C:\project\node_modules\webpack\lib\Compilation.js:758:12)
        at processModuleDependencies.err (C:\project\node_modules\webpack\lib\Compilation.js:697:9)
        at _combinedTickCallback (internal/process/next_tick.js:131:7)
        at process._tickCallback (internal/process/next_tick.js:180:9)
```
However, the file in question _types/IValidation, is imported in numerous other files without any warnings being emitted related to those other files that are also importing IValidation from _types/IValidation.

The contents of the file are simple, and there is an "export interface IValidation" in the file:
```javascript
export enum ValidationType {
    Required,
    Min,
    Max,
    MinLength,
    MaxLength,
}

export interface IValidation {
    validationType: ValidationType;
    errorMessage?: string;
    minValue?: number;
    maxValue?: number;
    maxLength?: number;
    minLength?: number;
}
```
We do have our "src" directory set as a module resolution root in the resolve section of our webpack config, but I also tried importing the file with a relative path and still get the warning.

I am kind of wondering if this is related to typescript.  We are using ts-loader, but we have it doing type checking only and are doing transpiling with babel.

**If the current behavior is a bug, please provide the steps to reproduce.**
Not sure since I get the warning due to the import of the file in question in one file but not due to several other files that also import the same interface from the same file.

**What is the expected behavior?**
It should not issue a warning about a missing export if the export clearly exists.

**Other relevant information:**
webpack version:  4.8.3
Node.js version: 8.9.1
Operating System: Windows 10
Additional tools: webpack-cli 2.1.3

**Comment by tryzniak** [comment]
Perhaps it would be awesome if you could create a simple project that would reproduce your bug and upload on github.

**Comment by sokra** [comment]
Interfaces no longer exist after transpiling. For webpack the export doesn't exist.

**Comment by domehead100** [comment]
True, typescript would remove the interfaces but should also remove the imports.

I'm not sure where the warning is coming from (maybe not from webpack itself).

I'll do some testing and see if I can figure it out.

We never had this happen on webpack 3.10 (were trying to upgrade to 4.8.3).

**Comment by Legends** [comment]
I guess, it's related to `ts-loader` then ...

**Comment by rgripper** [comment]
Here is a temp solution https://github.com/TypeStrong/ts-loader/issues/653#issuecomment-390889335

**Comment by johnnyreilly** [comment]
The temp solution suggested by @rgripper should work. I'd be open to baking this into ts-loader since it seems to have bitten a few people... Would anyone like to submit a PR?

**Comment by johnnyreilly** [comment]
On the linked thread I've put details, a suggestion for implementation and an offer of help if someone wants to submit a PR. Let's see what happens! :heart: :sunflower:

**Comment by sarod** [comment]
We are facing the same issue using babel-loader with https://babeljs.io/docs/en/babel-plugin-transform-typescript

**Comment by whitecolor** [comment]
The same problem here for TS types/interfaces after migrating from WP3 to WP4, multiple warnings:
```
VM606323 index.js:151 ./apps/client/state.ts 3:0-82
"export 'ChartElement' was not found in './cycles/ChartArea/chartElements'
```

I don't use `ts-loader` or anything special just simple loader:
```js
const ts = require('typescript')
const tsconfig = require('tsconfig')

const compilerOptions = tsconfig.loadSync(process.cwd()).config.compilerOptions

module.exports = function (source) {  
  const result = ts.transpile(source, { ...compilerOptions, ...this.query })
  return result
}
```
So it is not a problem of `ts-loader`

**Comment by igorkamyshev** [comment]
Hello. I create simple plugin for suppress this error. 👨‍💻

https://github.com/igorkamyshev/ignore-not-found-export-plugin

**Comment by anjunatl** [comment]
FWIW I only just started seeing this issue today in one of my react/redux/saga projects after refactoring my code layout from:

```
PageSagas.js
PageActions.js
PageReducers.js
```

to 

```
feature-A/sagas.js
feature-A/actions.js
feature-A/reducers.js
feature-B/sagas.js
feature-B/actions.js
feature-B/reducers.js
PageSagas.js
PageActions.js
PageReducers.js
```

to try to organize the growing project a bit more. **Not sure if the duplicate filenames across feature folders could have anything to do with it.**

And because more detail doesn't hurt with debugging, the exported function names don't match the filename. They currently look like:

```js
// feature-A/reducers.js :
export default function featureAReducer(...) {}
```

instead of

```js
// feature-A/reducers.js :
export default function reducers(...) {}
```

Hope this helps with the debugging. In the meantime, @igorkamyshev 's plugin looks good

**Comment by anjunatl** [comment]
Update: Some of the files the warnings fired for had both `export function X` and `export default function X` lines in them. All of the warnings went away when I removed all of the `default` keywords. Hope this helps.

**Comment by bodograumann** [comment]
@anjunatl, yours seems to be a different issue. This one is about exporting interfaces in typescript and is well understood. There is a comment somewhere explaining it really well, but I currently can not find it. Basically the typescript compiler emits no code for interfaces, so webpack can not find them in the compiled module; except when a module consists of only interfaces. In that case the module will end up completely empty and then webpack will not investigate the exports.

You should probably create a new issue for your problem.

**Comment by natew** [comment]
@igorkamyshev somehow, no matter using stats or your plugin, I'm still getting errors :/

**Comment by igorkamyshev** [comment]
> @igorkamyshev somehow, no matter using stats or your plugin, I'm still getting errors :/

Sorry, it was deprecated. =(

**Comment by elliotbonneville** [comment]
Is there currently a way to suppress these errors? I'm not having any luck with Webpack's `stat` config.

**Comment by natew** [comment]
For client at least you can: devServer: { clientLogLevel: 'error' } }

**Comment by elliotbonneville** [comment]
Thanks, @natew. Is there anything more specific? I'd like to leave other warnings intact.

**Comment by MarekLacoAXA** [comment]
Maybe this is not a bug, its a feature! (tm) :)
See:
https://github.com/Microsoft/TypeScript/issues/28481

When you export your TS interfaces/types in such a style so that it comply with TS's isolatedModules=true, then you won't get these webpack warnings.

```
import { State } from './interfaces';
export type State = State;
```

**Comment by pcx** [comment]
I think @MarekLacoAXA's comment nails the issue for me. I had this error because I was exporting types the same way I was exporting constants. I then split them and used `export type` for the type alone and it worked. Here's the code:

```ts
// ActionNames is a const and ActionTypes is a type

// Before
export { ActionNames, ActionTypes } from "./ActionTypes";

// Above line resulted in warning
// > "export 'ActionPayload' was not found in './ActionTypes'

// Changed the code to the following for the warning to disappear
export { ActionNames } from "./ActionTypes";
export type ActionPayload = ActionPayload;


```

**Comment by Emobe** [comment]
@pcx thank you. This actually fixed some huge problem I was having with barrel files. Basically just need to specify the interface to export rather than `export * from` because it does not compile so it has no idea where to look.

**Comment by dsebastien** [comment]
I have this issue as well. I've doubled checked my exports but I'm using the correct style.
I've tried multiple solutions but to no avail...

**Comment by tmtron** [comment]
@pcx can you please clarify your example? 
What is `AcitonPayload` and where it ist defined? 
Maybe you can edit the answer and show the relevant code in `ActionTypes`..

**Comment by garyo** [comment]
The only workaround I have found for this is to move all exported types and interfaces into a separate `types.ts` file and update all the imports everywhere to use that. It seems a pretty bad footgun that adding one class to a ts file that formerly happened to be all types and interfaces suddenly causes these warnings.
I'm not clear if the issue is with typescript (emitting only half of what webpack needs?) or with webpack being unable to tell that things are OK.

**Comment by tmtron** [comment]
When typescript transpiles the code to javascript, it knows about the import statements and it knows about the interfaces: so it could just remove the interface related import statements from the javascript-output, right?
If so, we should maybe file an issue for typescript?

**Comment by uqee** [comment]
Can confirm: moving types/interfaces away from any concrete stuff (into a separated file), is an ugly, but working workaround.

**Comment by david-golightly-leapyear** [comment]
This workaround apparently no longer works as of TS 3.7; I'm getting this error:

```
error TS2440: Import declaration conflicts with local declaration of 'FormFieldProps'.
```

However, you can workaround this by renaming the import:

```ts
import { FormFieldProps as _FormFieldProps } from './form-field-props'
export type FormFieldProps = _FormFieldProps
```

**Comment by drudv** [comment]
The solution proposed by MarekLacoAXA doesn't work if the exported type requires generics

**Comment by Anandnaik** [comment]
any concrete solution for this yet?

**Comment by bodograumann** [comment]
This will be fixed in typescript 3.8 by this PR: https://github.com/microsoft/TypeScript/pull/35200
Then you will be able to explicitely mark type imports like this (from the PR):

```
import type T from './mod';
import type { A, B } from './mod';
import type * as Types from './mod';

export type { T };
export type { T } from './mod';
```

Also compare the [announcement for typescript 3.8 beta](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/).

**Comment by johnnyreilly** [comment]
See also this comment by @andrewbranch: https://github.com/TypeStrong/ts-loader/issues/751#issuecomment-570954519

> We will probably recommend "preserve" or "error" for new projects going forward

This is in regards to the new (with TypeScript 3.8) compiler flag of `"importsNotUsedAsValue"`

**Comment by grreeenn** [comment]
Not sure if it happened to me for the same reason as to everyone else here, but in my case it happened because I used enum values as keys in my interface - like this:

```
export interface IDummyInterface {
    [eDummyEnum.A]: string;
}
```

Maybe it'll help someone

**Comment by JadeTao** [comment]
@grreeenn  same here. and I fixed by moving interfaces and enums into a seperate file.

**Comment by dsebastien** [comment]
With TypeScript 3.8, you can now use `import type { Blabla }`, which fixes the issue (the import statement is removed after transpilation.

**Comment by jeromeSH26** [comment]
Well the only solution I found is dirty in a way you need to separate the export of the components with the export of the interfaces... as mentionned by @david-golightly-leapyear . This adds usless code, especially when you have a lib with more than 200 interfaces to export,, but...

Running typescript 8.3 and webpack 4.4.2 

**details**
assuming a file `select.tsx` that creates a react component with interfaces describing props and the state.


1- at the end of component.tsx, exporting the component and its interfaces
`export { Select, ISelectProps, ISelectState };`

2- in the global index.ts, at the root of the project, importing all of them, but aliasing the interfaces
```
import {
	ISelectProps as _ISelectProps, _<---aliasing
	ISelectState as _ISelectState, _<--- aliasing
	Select,
} from "./UIComponents/Select";
```
3- then need to split the export in index.ts. 
Export the component on one side, the interfaces as type, using aliasing, on the other side


```
export {Select}
export type ISelectProps = _ISelectProps;
export type ISelectState = _ISelectState; 
```

**Comment by dsebastien** [comment]
For those interested, I've published a small article about this issue and how to fix it using TypeScript 3.8+: https://medium.com/@dSebastien/leveraging-type-only-imports-and-exports-with-typescript-3-8-5c1be8bd17fb

**Comment by monolithed** [comment]
There's a similar use case with old files:

File structure **before** refactoring:

```
provider.tsx
```

File structure **after** refactoring:
```
provider
   provider.tsx
   index.tsx
```

File structure after compilation (tsc):

```
provider
   provider.d.ts
   provider.js
   index.d.ts
   index.js
provider.d.ts
provider.js
```

As you can see `provider.d.ts` and `provider/index.d.ts` are both exists and the compiler uses the fist one.

**Comment by lonewarrior556** [comment]
A Truely ridiculous workaround that somehow work for us is 
```ts
// common/constants.ts
export const ACTIVE_FCIDS = ['Val1', 'Val2', 'Val3'] as const;
export type ACTIVE_FCID = 'Val1' |  'Val2' | 'Val3';
```

Issue: 
```ts
import { ACTIVE_FCID } from 'common/constants';
....
@Input() fcId: ACTIVE_FCID

// Failed to compile.
// Attempted import error: 'ACTIVE_FCID' is not exported from 'common/constants'
```
Work around:
```ts
import { ACTIVE_FCID } from 'common/constants';
....
@Input() fcId: ACTIVE_FCID | ACTIVE_FCID // Seriously!

// Compiled successfully!
```
@garyo solution also works if we seperate the `ACTIVE_FCIDS` and the type `ACTIVE_FCIDS` into separate files.

**Comment by michahell** [comment]
This also seems to occur in Angular projects with @angular/material, for instance:
```WARNING in ./libs/ui/action-menu/action-menu.component.ts 22:87-100
"export 'MenuPositionY' was not found in '@angular/material'```

if I add the exact file path where this type is defined in `.storybook/tsconfig.json`
`"files": ["./typings.d.ts", "../node_modules/@angular/material/menu.d.ts"]`
I still get this error

**Comment by lonewarrior556** [comment]
Try this https://github.com/webpack/webpack/issues/7378#issuecomment-617880809 this was an ang2 project.

**Comment by aboutqx** [comment]
Wierd, get this error when import from node modules,just js, not ts.

**Comment by tommyc38** [comment]
https://github.com/tommyc38/nest-pipe-repro

Above is repro...This issue is actually more than a warning and blows up when using class-validator in a mono repo.  Instructions are in the README, should be easy to reproduce but let me know if there are any issues.  You can see Storybook blowup while Angular actually doesn't have any issues.  Angular should run fine in Storybook.

@dsebastien thank you for your article!  Hopefully this helps solve the issue.

**Comment by tommyc38** [comment]
@lonewarrior556  , thought I would share a better way to code your example (in case you were unaware).
```ts
export const ACTIVE_FCIDS = <const> ['Val1', 'Val2', 'Val3'] ;
export type ACTIVE_FCID = typeof ACTIVE_FCIDS[number]
```
....now your code is dry.

**Comment by haexhub** [comment]
Hi,

I have a simliar problem in vue. 

_userDetails.ts_
```
class UserDetails implements IUserDetails {
  accountDisabled?: boolean
  accountExpiry?: Date
  username: string
  ...
}
export { UserDetails }
```

_userDetails.vue
```
<script lang="ts">
import Vue from 'vue'

import { UserDetails } from './userDetails'
...
</script>
```

Error message:

> "export 'UserDetails' was not found in './userDetails'

**Comment by ArseniyShestakov** [comment]
Yeah having the same issue, but weird part is that it's only affecting single interface import in one specific file and I simply can't come up with any ideas why.

**Comment by grreeenn** [comment]
Same here, popped right now. 
If my previous issue was caused by using enums as the interface keys, this time it's just a regular interface, one of the tens of others exported from the same file (generated). All of the other interfaces work just fine, this one gives the "export not found" warning (not an error, everything compiles and works)

**Comment by ryall** [comment]
As previously mentioned, `import type` or `export type` seems to fix this. Additionally, it's **only** a problem when **only importing a type and no concrete values**. 

```ts
// No error as we also import a concrete class
import { Class, IClass } from './class';

// Must mark with 'type' as we don't import a concrete class
import type { IClass } from './class';
```

**Comment by whitecolor** [comment]
For me this with TS using `export {} from` construct causes the issues.

```ts
import { AnimType } from './anim' // because need to use in this module
export { AnimType } from './anim' // because want it be avaialbled to import from this module 
```
This causes warning: `"export 'AnimType' was not found in './anim'`

To workaround:

```ts
import { AnimType as _AnimType } from './anim'
export type AnimType = _AnimType
```

**Comment by garyo** [comment]
Just use `import type {AnimType}...` instead. Requires Typescript 3.8.

**Comment by whitecolor** [comment]
> Just use `import type {AnimType}...` instead. Requires Typescript 3.8.

Actually, not for import but for export, this removes warning too:

```ts
export type { AnimType } from './anim'
```

This is still a workaround, and there is some bug in webpack.

**Comment by mirjalol-norkulov** [comment]
> For those interested, I've published a small article about this issue and how to fix it using TypeScript 3.8+: https://medium.com/@dSebastien/leveraging-type-only-imports-and-exports-with-typescript-3-8-5c1be8bd17fb

Thank you very much

**Comment by richnew10** [comment]
I found that this error appeared when some unused code was present deep in one of my Webpack targets, when that target included IE 10, and these two plugins were (redundantly, thanks to the IE target) present:

```
['@babel/transform-runtime', { regenerator: true }],
[
  '@babel/transform-regenerator',
  { generators: false, asyncGenerators: false, async: false },
],
```

Removing those two plugins resolved the issue.

**Comment by eng-dibo** [comment]
I got `"export 'resize' was not found in '@engineers/graphics'`

**graphics.ts**
https://github.com/eng-dibo/ngx/blob/master/packages/graphics/index.ts#L33
```
export type Img = string | Buffer | Sharp;
export type Size = number | string | null;
/**
 * [resize description]
 * @method resize
 * @param  {Img} img     file path or image buffer;
 * @param  {Size | Size[]} size    width or [width, height] or 'width,height', if size=null just convert img type
 * @param  {[type]} options [description]
 * @return Promise<info>
 *
 * todo:
 *  - rename size to dimensions
 *  - convert(img) = resize(img,size=null,{output=type})
 */
export function resize(
  img: Img,
  size: Size | Size[],
  options: ResizeOptions = {}
): Promise<any> {}
```

**in other file:**
https://github.com/eng-dibo/ngx/blob/master/projects/cms/server/api/v1.ts#L12
```
import { resize } from "@engineers/graphics/index";
```

**Comment by edaqa-uncountable** [comment]
The workaround forces the user to make a distinction between the types of the symbols being exported, which is not always possible nor necessary. Whether something is a class type, or an interface, enum, or type, shouldn't change how it is imported into the code.

Therefore I'd say the workaround is not a valid workaround. It requires making a change to valid code to satisfy an incorrect warning.

**Comment by lonewarrior556** [comment]
My definition of a "workaround" is `Alternative valid code that works and can replace the valid code that isn't currently working`

No one saying this is correct behavior (hence this ticket) however in the mean-time you can use the "workaround" to continue working until this is fixed.

**Comment by sokra** [comment]
Typescript introduced type-only exports/import which allows to generate valid JS code when compiling each file on it's own. Without type-only exports/import the TS to JS transformation can't be done when compiling each file on it's own. It's only possible with a full/partial graph of the modules. This is how typescript is designed. There is nothing to do on webpack-side here.

**Comment by severinbeauvais** [comment]
Re-exporting types in TS is coming soon!

https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/

**Comment by ndresx** [comment]
I just stumbled upon this issue and I'm wondering about a specific case where webpack/`HarmonyImportSpecifierDependency` seems to be unable to resolve a decorated + typed class member type, even though it works for another one that gets imported from the same file.

```ts
// types.ts
export type ComponentId = string;

export type ItemId = string;
```

```ts
// MyComponent.vue
import { Prop } from './decorators';
import { ComponentId, ItemId } from './types';

export default class MyComponent extends Vue {
  @Prop() readonly id!: ComponentId;

  generateItemId(): ItemId {
    return '123';
  } 
}
```

In this case, the `ComponentId` type "cannot be found", whereas the `ItemId` can. I'm really not sure what's going on here and why it works for one but not for the other - it doesn't look to me as if this would have something to do with how the import file is resolved.

```
"export 'ComponentId' was not found in './types'
```

Declaring another function with the `ComponentId` as return type doesn't change this behavior, only when the decorator itself is gone makes it properly work again.

**Comment by lonewarrior556** [comment]
> I just stumbled upon this issue and I'm wondering about a specific case where webpack/`HarmonyImportSpecifierDependency` seems to be unable to resolve a decorated + typed class member type, even though it works for another one that gets imported from the same file.
> 
> ```ts
> // types.ts
> export type ComponentId = string;
> 
> export type ItemId = string;
> ```
> 
> ```ts
> // MyComponent.vue
> import { Prop } from './decorators';
> import { ComponentId, ItemId } from './types';
> 
> export default class MyComponent extends Vue {
>   @Prop() readonly id!: ComponentId;
> 
>   generateItemId(): ItemId {
>     return '123';
>   } 
> }
> ```
> 
> In this case, the `ComponentId` type "cannot be found", whereas the `ItemId` can. I'm really not sure what's going on here and why it works for one but not for the other - it doesn't look to me as if this would have something to do with how the import file is resolved.
> 
> ```
> "export 'ComponentId' was not found in './types'
> ```
> 
> Declaring another function with the `ComponentId` as return type doesn't change this behavior, only when the decorator itself is gone makes it properly work again.

It has something to do with the decorator see if this works for you: https://github.com/webpack/webpack/issues/7378#issuecomment-617880809

In my project only the `@input()` variables showed this behavior this is definiltly a webpack issue and it's disappointing that it has been closed.

**Comment by Fydon** [comment]
I'm also experiencing this issue specifically with `@input()` variables, where the [#7378 (comment)](https://github.com/webpack/webpack/issues/7378#issuecomment-617880809) 
workaround works, although linting flags this. It would be helpful to address this case.

**Comment by archfz** [comment]
I got this warning once I upgraded to typescript from 3 to 4. Webpack is 4. From a first peak it seems to reproduce only in files that export only types / interfaces and also import some other types / interfaces from elsewhere.

> Typescript introduced type-only exports/import which allows to generate valid JS code when compiling each file on it's own. Without type-only exports/import the TS to JS transformation can't be done when compiling each file on it's own. It's only possible with a full/partial graph of the modules. This is how typescript is designed. There is nothing to do on webpack-side here.

@sokra I am not sure what is going on here. But I was on typescript `3.8.3` previously. As I understand type-only exports/import were introduced in typescript 3.8, so why do I reproduce this issues only once I upgrade to 4.x?

Since compilation works, I'll probably ignore this issue, unless someone has a solution for this.

**Comment by KurtPreston** [comment]
@andresesfm `import type` did it! Thank you!

**Comment by chuckntaylor** [comment]
@andresesfm I have an index.ts file which reexports the types. Using `export type { MyTypeOrInterface } from './MyTypeOrInterface'` did the trick. Thanks!

**Comment by elado** [comment]
Is there any way to treat these warnings as errors? I want my build to fail if it shows up. Currently both build and type check (`tsc --noEmit`) succeed so there's no way to catch this early.

**Comment by mgol** [comment]
@elado You can set `"importsNotUsedAsValues": "error"` in tsconfig: https://www.typescriptlang.org/tsconfig#importsNotUsedAsValues

That still allows mixed imports, consisting of both types & values. If you want to ensure types are only imported via `import type`, you can use the [`@typescript-eslint/consistent-type-imports` ESLint rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-imports.md).

**Comment by archfz** [comment]
Back. Now updated to webpack 5. So yeah, what about dynamic / start exports?

```
export * from './Service/IApiAuthProvider';
```

My generated types file.
```
declare namespace api {
    interface IApiAuthProvider {}
    class ApiAuthProvider {}
}
// other namespaces 
```

Imported in webpacked project as.
```
import {api} from 'my-module';
import IApiAuthProvider = api.IApiAuthProvider;
```

How should I handle this? As `export type * from` doesn't seem to be supported in typescript. See https://github.com/microsoft/TypeScript/issues/38306

**Comment by barclayd** [comment]
Adding the following will silence the warnings from webpack:

```javascript
module.exports = {
  module: {
    parser: {
      javascript: {
        reexportExportsPresence: 'false',
      },
    },
  },
};
```

**Comment by mwalkerr** [comment]
In my case I started getting these warnings after restructuring my project. The project is Spring Boot + React and I had previously kept the JS files under the `src/main/resources/static` folder. I moved them under a directory called `fe` and started getting errors like the ones mentioned here.

In my webpack config, I initially had:
```
resolve: {
...
    modules: [
      path.resolve(__dirname, "./src/main/resources/static/"),
      "node_modules",
    ]
  }
```
.. after moving everything in to `fe`, I updated that to:
```
resolve: {
...
    modules: [
      ".",
      "node_modules",
    ]
  }

```

.. which resulted in the warnings. I was able to prevent the warnings by updating that to 
```
resolve: {
...
    modules: [
      __dirname,
      "node_modules",
    ]
  }

```
No idea if this is correct or not but the warnings went away 🤷

**Comment by vankop** [comment]
@mwalkerr `resolve.modules` is path to `node_modules` directories ( by default only `node_modules`). You can several of them.

**Comment by neirongkuifa** [comment]
> ```js
> reexportExportsPresence
> ```

It should be:`reexportExportsPresence: false,`

**Comment by coler-j** [comment]
Is there a reason you would want `reexportExportsPresence` to be `true`?

**Comment by mpenney99** [comment]
The following works for me:

```
parser: {
  javascript: {
    importExportsPresence: false
  }
}
```
[https://webpack.js.org/configuration/module/#moduleparserjavascriptimportexportspresence]

I was getting this warning on all imports on types in my project after upgrading WebPack 4.41.6 to 5.74.0. It's a large project, so it would not be possible to refactor to type imports everywhere.

**Comment by ron23** [comment]
@mpenney99 actually you can use eslint to do it for you https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-imports.md

**Comment by ropotvs** [comment]
Try changing from `import { SomeInterface } from '../models';`

To `import type { SomeInterface } from '../models';`
